libraries/AP_Compass/AP_Compass_HMC5843.cpp

```diff
@@ -193,13 +193,13 @@ bool AP_Compass_HMC5843::init()
     // perform an initial read
     read();
 
-    _compass_instance = register_compass();
-
-    set_rotation(_compass_instance, _rotation);
-    
+    //register compass instance
     _bus->set_device_type(DEVTYPE_HMC5883);
+    _compass_instance = register_compass(_bus->get_bus_id());
```

Comment:
can we roll the set_dev_id() into the register_compass() call so we have less duplication?